fsck/fsck.c

```diff
@@ -685,6 +685,18 @@ static int check_name_dentry_set(struct exfat_de_iter *iter,
 		}
 	}
 
+	ret = exfat_check_name(inode->name, stream_de->stream_name_len);
+	if (ret != stream_de->stream_name_len) {
+		char err_msg[36];
+
+		snprintf(err_msg, sizeof(err_msg),
+			"invalid character '%c' in file name",
+			le16_to_cpu(inode->name[ret]));
+
+		return exfat_repair_rename_ask(&exfat_fsck, iter, inode->name,
```

Comment:
The error message below is a little confused. Can you append 'has', 'filename' or something else?

```
ERROR: '*' invalid character '*' in file name.
```

Reply:
I updated the error message to

```
ERROR: '*' filename has invalid character '*'.
```

Reply:
Thank you.
Looks good to me.